flow/connectors/postgres/postgres.go

```diff
@@ -797,6 +797,29 @@ func (c *PostgresConnector) EnsurePullability(
 	return &protos.EnsurePullabilityBatchOutput{TableIdentifierMapping: tableIdentifierMapping}, nil
 }
 
+func (c *PostgresConnector) ExportSnapshot(ctx context.Context) (string, any, error) {
+	var snapshotName string
+	tx, err := c.conn.Begin(ctx)
+	if err != nil {
+		return "", nil, err
+	}
+
+	err = tx.QueryRow(ctx, "select pg_export_snapshot()").Scan(&snapshotName)
```

Comment:
set `lock_timeout` and `idle_in_transaction_session_timeout` to 0 before returning, just like we do for slots